torch/quantization/fx/prepare.py

```diff
@@ -548,9 +548,10 @@ def maybe_insert_output_observer_for_node(
     is_standalone_module = qhandler is not None and \
         isinstance(qhandler, StandaloneModuleQuantizeHandler)
 
+    dtype = node_name_to_target_dtype[node.name]
     should_insert_observer = \
         qhandler.should_insert_observer_for_output(
-            qconfig, model.training)
+            qconfig, model.training) and dtype not in (torch.bool, None, torch.float)
```

Comment:
should we add dtype as an argument to `should_insert_observer_for_output`, so that the logic is in one place?

Reply:
yeah, that sounds good, will add in a follow up PR